### PR TITLE
[SPARK-49169][BUILD] Upgrade `commons-compress` to 1.27.0

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `commons-compress` from `1.26.2` to `1.27.0`. 

### Why are the changes needed?
The full release notes:
https://commons.apache.org/proper/commons-compress/changes-report.html#a1.27.0

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @dongjoon-hyun

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala [hunk 1]
```diff
@@ -760,7 +760,7 @@ abstract class AvroSuite
       assert(uncompressSize > deflateSize)
       assert(snappySize > deflateSize)
       assert(snappySize > bzip2Size)
-      assert(bzip2Size > xzSize)
+      assert(xzSize > bzip2Size)
```
Comment (dongjoon-hyun) [inline connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala:763]: Could you give us a pointer for this change, @panbingkun ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except one minor question)

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 4.0.0-preview2.

**Pull request closed without merging**